packages/cli/src/services/redis-client.service.ts

```diff
@@ -3,25 +3,62 @@ import ioRedis from 'ioredis';
 import type { Cluster, RedisOptions } from 'ioredis';
 import { Service } from 'typedi';
 
+import { Debounce } from '@/decorators/debounce';
 import { Logger } from '@/logger';
+import { TypedEmitter } from '@/typed-emitter';
 
 import type { RedisClientType } from '../scaling/redis/redis.types';
 
+type RedisEventMap = {
+	'connection-lost': number;
+	'connection-recovered': never;
+};
+
 @Service()
-export class RedisClientService {
+export class RedisClientService extends TypedEmitter<RedisEventMap> {
 	private readonly clients = new Set<ioRedis | Cluster>();
 
+	private readonly config = {
+		/** How long (in ms) to try to reconnect for before exiting. */
+		maxTimeout: this.globalConfig.queue.bull.redis.timeoutThreshold,
+
+		/** How long (in ms) to wait between reconnection attempts. */
+		retryInterval: 1000,
+
+		/** How long (in ms) to wait before resetting the cumulative timeout. */
+		resetLength: 30_000,
+
+		/** Whether any client has lost connection to Redis. */
+		lostConnection: false,
```

Comment:
I'm not sure what I feel about storing state in `config`. Maybe we could rename this to something like `connectionState` or move the state outside the config?

Reply:
Please see [0417d2e](https://github.com/n8n-io/n8n/pull/11007/commits/0417d2ee671bc111ec9369635e40d13632b8b335)